services/manifest-repo-export-service/pkg/repository/repository.go

```diff
@@ -354,19 +355,39 @@ func DefaultPushActionCallback(pushOptions git.PushOptions, r *repository) PushA
 
 type PushUpdateFunc func(string, *bool) git.PushUpdateReferenceCallback
 
-func (r *repository) ProcessQueueOnce(ctx context.Context, t Transformer, callback PushUpdateFunc, pushAction PushActionCallbackFunc, tx *sql.Tx) error {
+func (r *repository) ProcessQueueOnce(ctx context.Context, t Transformer, tx *sql.Tx) error {
 	span, ctx := tracer.StartSpanFromContext(ctx, "ProcessQueueOnce")
 	defer span.Finish()
 
 	log := logger.FromContext(ctx).Sugar()
 
-	var pushSuccess = true
+	// Apply the items
+	apply := func() (error, *TransformerResult) {
+		err, changes := r.applyTransformerBatches(ctx, t, true, tx)
+		if err != nil {
+			log.Warnf("rolling back transaction because of %v", err)
+			return err, nil
+		}
+		return nil, changes
+	}
 
+	err, _ := apply()
+	if err != nil {
+		return fmt.Errorf("first apply failed, aborting: %v", err)
+	}
+	return nil
+}
+
+func (r *repository) PushRepo(ctx context.Context) error {
+	//if true {
```

Comment:
Remove these commented lines if they're not needed.

Reply:
done